src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationValidationExecutor.cs

```diff
@@ -0,0 +1,104 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Microsoft.IdentityModel.Logging;
+
+namespace Microsoft.IdentityModel.Protocols
+{
+    /// <summary>
+    /// Validates configuration by using validator.
+    /// </summary>
+    /// <typeparam name="T">The type of the configuration.</typeparam>
+    public class ConfigurationValidationExecutor<T> where T : class
+    {
+        /// <summary>
+        /// Instantiates a new <see cref="ConfigurationValidationExecutor{T}"/> that validate retrived configuration data.
+        /// </summary>
+        /// <param name="defaultConfigurationValidator">The default configuration validator.</param>
+        public ConfigurationValidationExecutor(IConfigurationValidator<T> defaultConfigurationValidator)
+        {
+            DefaultConfigurationValidator = defaultConfigurationValidator ?? throw LogHelper.LogArgumentNullException(nameof(defaultConfigurationValidator));
+        }
+
+        /// <summary>
+        /// Instantiates a new <see cref="ConfigurationValidationExecutor{T}"/> that validate retrived configuration data.
+        /// </summary>
+        /// <param name="defaultConfigurationValidator">The default configuration validator.</param>
+        /// <param name="additionalConfigurationValidator">The additional configuration validator.</param>
+        public ConfigurationValidationExecutor(IConfigurationValidator<T> defaultConfigurationValidator, IConfigurationValidator<T> additionalConfigurationValidator)
+            : this(defaultConfigurationValidator)
+        {
+            AdditionalConfigurationValidator = additionalConfigurationValidator ?? throw LogHelper.LogArgumentNullException(nameof(additionalConfigurationValidator));
+        }
+
+        /// <summary>
+        /// The additional Configuration Validator.
+        /// </summary>
+        public IConfigurationValidator<T> AdditionalConfigurationValidator { get; }
```

Comment:
If this is for chaining, then another pattern is to have the IConfigurationValidator expose an 'Inner' property of type IConfigurationValidator<T>.

Why do we need chaining?